src/app/clusters/testing/ClusterTesting.h

```diff
@@ -0,0 +1,97 @@
+/*
+ *    Copyright (c) 2025 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app/AttributeValueDecoder.h>
+#include <app/AttributeValueEncoder.h>
+#include <app/ConcreteAttributePath.h>
+#include <app/data-model-provider/tests/ReadTesting.h>
+#include <app/data-model-provider/tests/WriteTesting.h>
+#include <app/server-cluster/ServerClusterInterface.h>
+#include <lib/core/TLVReader.h>
+#include <memory>
+
+namespace chip {
+namespace Test {
+
+// Helper class for testing clusters.
+// Currently it has a generic ReadAttribute() and WriteAttribute().
+// It can be extended in the future to support other cluster operations.
+// The function template instantiations will succeed only for types that support the expected operations.
+// For example, in the case of ReadAttribute(), the call to app::DataModel::Decode() for the given type T should be supported.
+// Otherwise, that template instantiation will fail.
```

Comment:
Unsure if we should say all the `This only works for type that works the expected operation` ... this seems like very vague (what supported operation is expected? It seems obvious things only work when template instantiation works).

We could say that Read/Write of all attribute types should work, but make sure you use `::Type` for encoding and `DecodableType` for decoding for structure types.